src/mam4xx/lin_strat_chem.hpp

```diff
@@ -0,0 +1,322 @@
+    const ColumnView &o3_vmr,
+    // diagnostic variables outputs
+    const ColumnView &do3_linoz, const ColumnView &do3_linoz_psc,
+    const ColumnView &ss_o3, const ColumnView &o3col_du_diag,
+    const ColumnView &o3clim_linoz_diag, const ColumnView &sza_degrees) {
+
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, ltrop), KOKKOS_LAMBDA(int kk) {
+        lin_strat_chem_solve_kk(
+            o3col(kk), temperature(kk), sza, pmid(kk), delta_t, rlats,
+            // ltrop, & !in
+            linoz_o3_clim(kk), linoz_t_clim(kk), linoz_o3col_clim(kk),
+            linoz_PmL_clim(kk), linoz_dPmL_dO3(kk),
+            linoz_dPmL_dT(kk), // in
+            linoz_dPmL_dO3col(kk),
+            linoz_cariolle_psc(kk), // in
+                                    //
+            chlorine_loading,
+            psc_T, // PSC ozone loss T (K) threshold
+            o3_vmr(kk),
+            // diagnostic variables outputs
+            do3_linoz(kk), do3_linoz_psc(kk), ss_o3(kk), o3col_du_diag(kk),
+            o3clim_linoz_diag(kk), sza_degrees(kk));
+      });
+
+} // lin_strat_chem_solve
+
+KOKKOS_INLINE_FUNCTION
+void lin_strat_sfcsink(const Real delta_t, const ColumnView &pdel, // in
+                       const ColumnView &o3l_vmr, const Real o3_sfc,
+                       const int o3_lbl, const Real o3_tau, Real &o3l_sfcsink) {
+
+  // @param[in] delta_t              timestep size [secs]
+  // @param[in] pdel(ncol,pver)      pressure delta about midpoints [Pa]
+
+  // !inten in-outs
+  // @param[inout]:: o3l_vmr(ncol ,pver)             ! ozone volume mixing ratio
+  // [vmr]
+
+  constexpr Real one = 1.0;
+  // BAD CONSTANT
+  constexpr Real mwo3 = 48.; // molecular weight O3
+  constexpr Real mwdry = haero::Constants::molec_weight_dry_air *
+                         1e3; //     ! molecular weight dry air ~ kg/kmole;//!
+                              //     molecular weight dry air
+  constexpr Real rgrav =
+      one / haero::Constants::gravity; // reciprocal of gravit
+
+  const Real efactor =
+      (one - haero::exp(-delta_t / o3_tau)); // !compute time scale factor
+  Real do3mass_icol = 0;
+  for (int kk = pver - 1; kk > pver - o3_lbl - 1; --kk) {
+    const Real mass = pdel(kk) * rgrav; //   air mass in kg/m2
+
+    const Real o3l_old = o3l_vmr(kk);              // vmr
+    const Real do3 = (o3_sfc - o3l_old) * efactor; // vmr
+    const Real o3l_new = o3l_old + do3;
+    do3mass_icol +=
+        do3 * mass * mwo3 / mwdry; //   loss in kg/m2 summed over boundary
+                                   //   layers within one time step
+    o3l_vmr(kk) = o3l_new;
+  }
+
+  // Two parameters are applied to Linoz O3 for surface sink, O3l is not coupled
+  // to real O3
+  constexpr Real KgtoTg = 1.0e-9; //
+  constexpr Real peryear =
+      86400. * 365.0; // ! to multiply to convert per second to per year
+
+  o3l_sfcsink =
+      do3mass_icol / delta_t * KgtoTg * peryear; // ! saved in Tg/yr/m2 unit
+
+} // lin_strat_sfcsink
```

Comment:
This is another one where the comment doesn't make sense to me. where's the `m^-2` unit?

As calculated, looks like [mass / time] or [Tg / yr]

Reply:
And this is picky, but I personally think it invites confusion when math is written in the left-to-right code execution order of operations and would favor keeping numerator and denominator together.
```
years_per_second = pow(persecond_badname, -1);
o3l_sfcsink =  (do3mass_icol * KgtoTg) / (delta_t * years_per_second);
```
No need to actually change, but maybe one day in the future...

Reply:
> This is another one where the comment doesn't make sense to me. where's the `m^-2` unit?
> 
> As calculated, looks like [mass / time] or [Tg / yr]

I guess m^-2 is from `do3mass_icol`, is this correct @brigaudet or @singhbalwinder ? 
`do3mass_icol` => [kg/m^2]

Reply:
> And this is picky, but I personally think it invites confusion when math is written in the left-to-right code execution order of operations and would favor keeping numerator and denominator together.
> 
> ```
> years_per_second = pow(persecond_badname, -1);
> o3l_sfcsink =  (do3mass_icol * KgtoTg) / (delta_t * years_per_second);
> ```
> 
> No need to actually change, but maybe one day in the future...

I believe this  is:
`o3l_sfcsink(:ncol) = (do3mass(:ncol)/delta_t) * KgtoTg * peryear ! saved in Tg/yr/m2 unit`

Reply:
> > This is another one where the comment doesn't make sense to me. where's the `m^-2` unit?
> > As calculated, looks like [mass / time] or [Tg / yr]
> 
> I guess m^-2 is from `do3mass_icol`, is this correct @brigaudet or @singhbalwinder ? `do3mass_icol` => `[kg/m^2]`

Yes, [mass] = [pdel] x [rgrav] = (kg/(s^2-m)) (s^2/m) = kg/m^2.  do3 and mwo3/mwdry are dimensionless.  Thus 'do3mass' has the same units as 'mass', kg/m^2.